From 23e6f94591e72a246183810549561296ca32bf9a Mon Sep 17 00:00:00 2001
From: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date: Wed, 28 Oct 2020 23:02:58 -0400
Subject: [PATCH] Allow toybox binary to have a prefix too

This allows the toybox binary to work once placed in the Nix store, as
elements in the Nix store are prefixed with a hash.
---
 main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main.c b/main.c
index 7c60bdf4..50cd57ed 100644
--- a/main.c
+++ b/main.c
@@ -30,7 +30,7 @@ struct toy_list *toy_find(char *name)
   if (!CFG_TOYBOX || strchr(name, '/')) return 0;
 
   // Multiplexer name works as prefix, else skip first entry (it's out of order)
-  if (!toys.which && strstart(&name, "toybox")) return toy_list;
+  if (!toys.which && strstr(name, "toybox")) return toy_list;
   bottom = 1;
 
   // Binary search to find this command.
-- 
2.28.0

